docs/reference/packages.rst

```diff
@@ -1,314 +1,150 @@
+
+``mip`` ("mip installs packages") is similar in concept to Python's ``pip`` tool,
+however it does not use the PyPi index, rather it uses :term:`micropython-lib`
+as its index by default. ``mip`` will automatically fetch compiled
+:term:`.mpy file` when downloading from micropython-lib.
+
+The most common way to use ``mip`` is from the REPL::
+
+    >>> import mip
+    >>> mip.install("pkgname")  # Installs the latest version of "pkgname" (and dependencies)
+    >>> mip.install("pkgname", version="x.y")  # Installs version x.y of "pkgname"
+    >>> mip.install("pkgname", mpy=False)  # Installs the source version (i.e. .py rather than .mpy files)
+
+``mip`` will detect an appropriate location on the filesystem by searching
+``sys.path`` for the first entry ending in ``/lib``. You can override the
+destination using ``target``, but note that this path must be in ``sys.path`` to be
+able to subsequently import it.::
+
+    >>> mip.install("pkgname", target="third-party")
+    >>> sys.path.append("third-party")
+
+As well as downloading packages from the micropython-lib index, ``mip`` can also
+install third-party libraries. The simplest way is to download a file directly::
+
+    >>> mip.install("http://example.com/x/y/foo.py")
+    >>> mip.install("http://example.com/x/y/foo.mpy")
+
+When installing a file directly, the ``target`` argument is still supported to set
+the destination path, but ``mpy`` and ``version`` are ignored.
+
+The URL can also start with ``github:`` as a simple way of pointing to content
+hosted on GitHub::
+
+    >>> mip.install("github:org/repo/path/foo.py")  # Uses default branch
+    >>> mip.install("github:org/repo/path/foo.py", version="branch-or-tag")  # Optionally specify the branch or tag
+
+More sophisticated packages (i.e. with more than one file, or with dependencies)
+can be downloaded by specifying the path to their ``package.json``.
+
+    >>> mip.install("http://example.com/x/package.json")
+    >>> mip.install("github:org/user/path/package.json")
+
+If no json file is specified, then "package.json" is implicitly added::
+
+    >>> mip.install("http://example.com/x/")
+    >>> mip.install("github:org/repo")
+    >>> mip.install("github:org/repo", version="branch-or-tag")
+
+
+Using ``mip`` on the Unix port
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On the Unix port, ``mip`` can be used at the REPL as above, and also by using ``-m``::
+
+    $ ./micropython -m mip install pkgname-or-url
+    $ ./micropython -m mip install pkgname-or-url@version
+
+The ``--install-target=path`` and ``--no-install-mpy`` arguments can be set::
+
+    $ ./micropython -m mip install --install-target=third-party pkgname
+    $ ./micropython -m mip install --install-target=third-party-source --no-install-mpy pkgname
+
+Installing packages with ``mpremote``
+-------------------------------------
+
+The :term:`mpremote` tool also includes the same functionality as ``mip`` and can
+be used to install packages to a connected device::
```

Comment:
maybe make it clear that mpremote runs on the host and installs on the target, and so the target doesn't need to have networking support